src/script/interpreter.cpp

```diff
@@ -1525,7 +1525,7 @@ bool VerifyScript(const CScript& scriptSig, const CScript& scriptPubKey, const C
     //LogPrintf(">>>>VerifyScript true\n");
     if (stack.empty())
         return set_error(serror, SCRIPT_ERR_EVAL_FALSE);
-    if (CastToBool(stack.back()) == false)
+    if (CastToBool(stack.back()))
```

Comment:
Needs a ! otherwise these two lines are no longer equivalent.

Reply:
Woah, good catch, I don't know how I missed that, but I did.